django/db/models/sql/query.py

```diff
@@ -1007,10 +1008,10 @@ def prepare_lookup_value(self, value, lookups, can_reuse, allow_joins=True):
         # Subqueries need to use a different set of aliases than the
         # outer query. Call bump_prefix to change aliases of the inner
         # query (the value).
-        if hasattr(value, 'query') and hasattr(value.query, 'bump_prefix'):
-            value = value._clone()
-            value.query.bump_prefix(self)
-        if hasattr(value, 'bump_prefix'):
+        if hasattr(value, '_prepare_as_filter_value'):
+            value = value._prepare_as_filter_value()
+            value.bump_prefix(self)
+        elif hasattr(value, 'bump_prefix'):
```

Comment:
maybe we could remove this branch, and have _prepare_as_filter_value for all objects that have bump_prefix defined?